include/linux/soundwire/sdw.h

```diff
@@ -950,10 +950,12 @@ void sdw_show_ping_status(struct sdw_bus *bus, bool sync_delay);
  *
  * @num: Port number
  * @ch_mask: channels mask for port
+ * @lane_mask: Which lanes are available
  */
 struct sdw_port_config {
 	unsigned int num;
 	unsigned int ch_mask;
+	unsigned int lane_mask;
```

Comment:
````lane 0 can only be used by port 1````

I don't think it's realistic to assume that some ports can only be used with non-zero lanes. By default I think the behavior is that all ports support Lane0. It'd be surprised if this wasn't the case, since that's the common denominator for all systems. It's possible that some non-zero lanes are not supported, e.g. a device could perfectly well support lane0..3 only.

````
The lane_maps[i] will be 0 by default and we can't tell if a lane is connected to lane 0 or not connected.
````

I couldn't follow the "lane is connected to lane0"....